packages/block-editor/src/components/use-block-sync/index.js

```diff
@@ -70,8 +86,10 @@ export default function useBlockSync( {
 	selectionEnd: controlledSelectionEnd,
 	onChange = noop,
 	onInput = noop,
+	entityId,
 } ) {
 	const registry = useRegistry();
+	const instanceId = useRef( uuid() ).current;
```

Comment:
The `withInstanceId`/`useInstanceId` won't work here because they rely on mapping an object (e.g. component) to a new ID. This approach doesn't work with custom hooks directly. (it might be useful to move this logic to that package so that it does, though)